keras_cv/models/feature_extractor/clip/clip_model.py

```diff
@@ -180,8 +157,52 @@ def call(self, inputs):
             * logit_scale
         )
         logits_per_text = ops.transpose(logits_per_image)
+        outputs = {
+            "logits_per_image": logits_per_image,
+            "logits_per_text": logits_per_text,
+        }
+
+        super().__init__(
+            inputs={
+                "image_input": image_input,
+                "text_input": text_input,
+                "attention_mask_input": attention_mask_input,
+            },
+            ouputs=outputs,
+            **kwargs,
+        )
+        self.embed_dim = embed_dim
+        self.image_resolution = image_resolution
+        self.vision_layers = vision_layers
+        self.vision_width = vision_width
+        self.vision_patch_size = vision_patch_size
+        self.context_length = context_length
+        self.vocab_size = vocab_size
+        self.transformer_width = transformer_width
+        self.transformer_heads = transformer_heads
+        self.transformer_layers = transformer_layers
 
-        return logits_per_image, logits_per_text
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.text_encoder.build([None, self.context_length])
+        self.image_encoder.build(
+            [None, self.image_resolution, self.image_resolution, 3]
+        )
```

Comment:
I think you can remove the build and compute output shape method now. Only layers/models that are not implemented in the functional manner need them.